src/Utilities/Blaze.hpp

```diff
@@ -41,4 +37,77 @@ constexpr bool useDefaultInitialization = false;
 
 // Disable HPX parallelization
 #define BLAZE_HPX_PARALLEL_MODE 0
+
+// Disable all padding
+#define BLAZE_USE_PADDING 0
+
+#define BLAZE_USE_STREAMING 1
+#define BLAZE_USE_OPTIMIZED_KERNELS 1
+#define BLAZE_USE_DEFAULT_INITIALIZATON 0
```

Comment:
So, I think the suggestion of using compile flags is the 'right' way to use these blaze flags, but our choices in cmake make this difficult for us. In particular, any compile definitions that are set outside the pre-compiled header must similarly be set for every compilation that used the pre-compiled header. I was unable to find a clear way to propagate those compile definitions in a straightforward way that did not result in a `-winvalid-pch` from gcc when using a pre-compiled header.
I'm open to suggestions, but because this is how we've handled blaze up untill now, my inclination is to enforce the include order consistently in our source files and include a code comment where applicable, but leave the cmake best practices aside for now, particularly because (based on our pch handling) the cmake pre-compiled header support seems somewhat incomplete.

Reply:
I think the PCH lib just needs to link `PUBLIC`ly with `Blaze`. This seems to work on develop in the container with gcc-9:

```diff
diff --git a/cmake/SetupBlaze.cmake b/cmake/SetupBlaze.cmake
index d35a636fc..6f1396805 100644
--- a/cmake/SetupBlaze.cmake
+++ b/cmake/SetupBlaze.cmake
@@ -17,6 +17,22 @@ set_property(TARGET Blaze PROPERTY
 set_property(TARGET Blaze PROPERTY
   INTERFACE_LINK_LIBRARIES Lapack)
 
+target_compile_definitions(
+  Blaze
+  INTERFACE
+  # Override SMP configurations
+  _BLAZE_SYSTEM_SMP_H_
+  BLAZE_USE_SHARED_MEMORY_PARALLELIZATION=0
+  BLAZE_OPENMP_PARALLEL_MODE=0
+  BLAZE_CPP_THREADS_PARALLEL_MODE=0
+  BLAZE_BOOST_THREADS_PARALLEL_MODE=0
+  # Disable MPI parallelization
+  _BLAZE_SYSTEM_MPI_H_
+  BLAZE_MPI_PARALLEL_MODE=0
+  # Disable HPX parallelization
+  BLAZE_HPX_PARALLEL_MODE=0
+  )
+
 add_interface_lib_headers(
   TARGET Blaze
   HEADERS
diff --git a/cmake/SetupPch.cmake b/cmake/SetupPch.cmake
index 5313d6a3a..c9f4fe273 100644
--- a/cmake/SetupPch.cmake
+++ b/cmake/SetupPch.cmake
@@ -177,7 +177,7 @@ if (USE_PCH)
     )
   target_link_libraries(
     ${SPECTRE_PCH_LIB}
-    PRIVATE
+    PUBLIC
     Blaze
     Brigand
     )
diff --git a/src/Utilities/Blaze.hpp b/src/Utilities/Blaze.hpp
index 5d8b11bfb..02dd3ef87 100644
--- a/src/Utilities/Blaze.hpp
+++ b/src/Utilities/Blaze.hpp
@@ -27,18 +27,4 @@ constexpr bool useOptimizedKernels = true;
 namespace blaze {
 constexpr bool useDefaultInitialization = false;
 }
-
-// Override SMP configurations
-#define _BLAZE_SYSTEM_SMP_H_
-#define BLAZE_USE_SHARED_MEMORY_PARALLELIZATION 0
-#define BLAZE_OPENMP_PARALLEL_MODE 0
-#define BLAZE_CPP_THREADS_PARALLEL_MODE 0
-#define BLAZE_BOOST_THREADS_PARALLEL_MODE 0
-
-// Disable MPI parallelization
-#define _BLAZE_SYSTEM_MPI_H_
-#define BLAZE_MPI_PARALLEL_MODE 0
-
-// Disable HPX parallelization
-#define BLAZE_HPX_PARALLEL_MODE 0
 /// \endcond
```

Reply:
On a bare build directory, the patch you posted generates gcc warnings:
```
cc1plus: warning: /home/mox/research/SXS/spectre/spectre-build-fresh/SpectrePch.hpp.gch: not used because `BLAZE_BOOST_THREADS_PARALLEL_MODE' not defined [-Winvalid-pch]
```
my cmake call was:
```
cmake -D CHARM_ROOT=/work/charm_6_10_2/multicore-linux-x86_64-gcc/ -D CMAKE_CXX_COMPILER=/usr/bin/g++-9 -D CMAKE_C_COMPILER=/usr/bin/gcc-9 -D CMAKE_Fortran_COMPILER=/usr/bin/gfortran-7 -D USE_PCH=ON /home/mox/research/SXS/spectre/
```
(critically, I think it is necessary to force a rebuild of your PCH to see the behavior -- I needed to clear my build directory)